### PR TITLE
fix: Remove non-operational bridges and add stats.

### PR DESCRIPTION
**Pull request by bgrozev**

### CODE REVIEW

File: src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java [hunk 1]
```diff
@@ -1575,6 +1575,7 @@ public XMPPError onSessionInfo(JingleSession session, JingleIQ iq)
                     address,
                     bridgeSessionId));
         }
+        listener.participantIceFailed();
```
Comment (bbaldino) [inline src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java:1578]: Should we update this even in the "invalid session" case? (I'm not sure what 'invalid session' implies, exactly)

**Review by bbaldino — DISMISSED** [review]
One small question, but lgtm.

File: src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java (hunk 1 above)
Comment (bgrozev) [reply src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java:1578]: I don't know what the "invalid session" case is either. I wanted to keep the semantics closer to "how many times a client reported ICE failed", so this made sense.

File: src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java [hunk 2]
```diff
@@ -870,14 +884,25 @@ private void inviteParticipant(
             boolean[] startMuted)
     {
         BridgeSession bridgeSession;
+
+        // Some of the bridge in the conference may have become non-operational. Inviting a new participant to the
```
Comment (JonathanLennox) [inline src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java:888]: Grammar: "some of the bridges"

File: src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java [hunk 3]
```diff
@@ -846,8 +846,22 @@ private Bridge selectBridge(Participant participant)
...
+    /**
+     * Removes all non-operational bridges from the conference and re-invites their participants.
+     */
+    private void removeNonOperationalBridges()
+    {
+        Set<Jid> nonOperationalBridges = bridges.stream()
+                .filter(session -> session.hasFailed || !session.bridge.isOperational())
+                .map(session -> session.bridge.getJid())
+                .collect(Collectors.toSet());
+        if (nonOperationalBridges.size() > 0)
```
Comment (JonathanLennox) [inline src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java:861]: Suggest `if (!nonOperationalBridges.empty())`.

File: src/main/java/org/jitsi/jicofo/stats/Statistics.java [hunk 4]
```diff
@@ -18,4 +18,24 @@
      * Jicofo since it was started
      */
     public final AtomicInteger totalParticipants = new AtomicInteger(0);
+
+    /**
+     * The number of participants that were moved away from a failed bridge.
+     */
+    public final AtomicInteger totalParticipantsMoved = new AtomicInteger();
```
Comment (JonathanLennox) [inline src/main/java/org/jitsi/jicofo/stats/Statistics.java:25]: Suggest these use `LongAdder`?  But probably all the stats should be updated at once.

Comment (bgrozev) [reply src/main/java/org/jitsi/jicofo/stats/Statistics.java:25]: I don't think any of these are updated often enough for performance to be a concern.

**Review by JonathanLennox — APPROVED** [review]

**Pull request merged**